mlos_bench/mlos_bench/config/schemas/optimizers/optimizer-schema.json

```diff
@@ -201,6 +203,46 @@
                 }
             },
             "else": false
+        },
+
+        {
+            "$comment": "extensions to the 'config' object properties when the manual optimizer is being used",
+            "if": {
+                "properties": {
+                    "class": {
+                        "enum": [
+                            "mlos_bench.optimizers.ManualOptimizer",
+                            "mlos_bench.optimizers.manual_optimizer.ManualOptimizer"
+                        ]
+                    }
+                },
+                "required": ["class"]
+            },
+            "then": {
+                "properties": {
+                    "config": {
+                        "type": "object",
+                        "allOf": [
+                            { "$ref": "#/$defs/config_base_optimizer" },
+                            {
+                                "type": "object",
+                                "properties": {
+                                    "cycle_tunable_values": {
+                                        "description": "The tunable values to cycle through.",
+                                        "type": "array",
+                                        "items": {
+                                            "$ref": "../tunables/tunable-values-schema.json#/$defs/tunable_values_set"
+                                        }
+                                    }
+                                }
+                            }
+                        ],
+                        "$comment": "disallow other properties",
+                        "unevaluatedProperties": false
+                    }
+                }
+            },
+            "else": false
```

Comment:
Future work, but we should probably split this schema up into smaller files like we do for the other schemas.

Reply:
https://github.com/microsoft/MLOS/issues/867